hns/hns_test.go

```diff
@@ -360,6 +371,109 @@ var _ = Describe("HNS wrapper", func() {
 	})
 })
 
+var _ = Describe("HNS race conditions workarounds", func() {
+
+	var targetAddr string
+	const (
+		numTries = 20
+	)
+
+	BeforeEach(func() {
+		if !useActualController {
+			Skip("useActualController flag is false. Won't perform HNS race conditions test.")
+		}
+
+		targetAddr = fmt.Sprintf("%s:%v", controllerAddr, controllerPort)
+		err := common.HardResetHNS()
+		Expect(err).ToNot(HaveOccurred())
+	})
+
+	Specify("without HNS networks, connections work", func() {
+		_, err := net.Dial("tcp", targetAddr)
+		Expect(err).ToNot(HaveOccurred())
+	})
+
+	Context("subnet is specified in new HNS switch config", func() {
+
+		subnets := []hcsshim.Subnet{
+			{
+				AddressPrefix:  "10.0.0.0/24",
+				GatewayAddress: "10.0.0.1",
+			},
+		}
+		configuration := &hcsshim.HNSNetwork{
+			Type:               "transparent",
+			NetworkAdapterName: "Ethernet0",
```

Comment:
Is there a way to programatically get network adapter name, rather that hardcoding it?